node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,612 @@
+		topics.ForEach(func(topicKey, topicValue gjson.Result) bool {
+			if topicValue.String() == TOPIC_LOG_MSG {
+				logger.Info("RO:  Found topic")
+
+				// data is inside the logs object
+				logDataBase := gjson.Get(logString, "data")
+				if !logDataBase.Exists() {
+					logger.Error("Message has no data field", zap.String("txhash", txhash))
+					return false
+				}
+				// use the address inside the logs object
+				emitterBase := gjson.Get(logString, "address")
+				if !emitterBase.Exists() {
+					logger.Error("Message has no address field", zap.String("txhash", txhash))
+					return false
+				}
+
+				emitter := emitterBase.String()
+				emitterAddr, err := vaa.StringToAddress(emitter)
+				if err != nil {
+					logger.Fatal("failed to unpack emitter address", zap.Error(err))
+					return false
+				}
+
+				// Get the other values from the Data value
+				logDataString := logDataBase.String()
+				logDataString = logDataString[2:] // remove the leading 0x
+				logDataBytes, err := hex.DecodeString(logDataString)
+				if err != nil {
+					logger.Fatal("failed to unpack log data", zap.Error(err))
+					return false
+				}
+
+				unpackedMsg, err := ethAbi.Unpack("LogMessagePublished", logDataBytes)
+				if err != nil {
+					logger.Fatal("failed to unpack log data", zap.Error(err))
+					return false
+				}
+
+				seq := unpackedMsg[0].(uint64)
+				nonce := unpackedMsg[1].(uint32)
+				payload := unpackedMsg[2].([]byte)
+				cLevel := unpackedMsg[3].(uint8)
+				logger.Info("unpackedMsg",
+					zap.Int("Length of unpackedMsg", len(unpackedMsg)),
+					zap.Uint64("Sequence", seq),
+					zap.Uint32("Nonce", nonce),
+					zap.Uint8("consistencyLevel", cLevel))
+				var chainID vaa.ChainID = 27
+
+				messagePublication := &common.MessagePublication{
+					TxHash:           txHash,      // In log
+					Timestamp:        unixTS,      // In log
+					Nonce:            nonce,       // In log data
+					Sequence:         seq,         // In log data
+					EmitterChain:     chainID,     // Don't know where to get this
+					EmitterAddress:   emitterAddr, // In log
+					Payload:          payload,     // In log data
+					ConsistencyLevel: cLevel,      // In log data
+				}
+
+				logger.Info("messagePublication",
+					zap.Stringer("txHash", txHash),
+					zap.Stringer("timestamp", unixTS),
+					zap.Uint32("nonce", nonce),
+					zap.Uint64("sequence", seq),
+					zap.Stringer("EmitterAddr", emitterAddr),
+					zap.Uint8("ConsistenceLevel", cLevel),
+				)
+
+				events = append(events, messagePublication)
+				blockNum := gjson.Get(msg, "block_number")
+				logger.Info("RO: Found True Log Msg", zap.Stringer("block Number", blockNum))
+				return false // break out of the ForEach() loop
+			}
+			return true // continue inner ForEach() loop
+		})
+		return true // continue outer ForEach() loop
+	})
+
+	return events, nil
+}
+
+// StringToAddress convert string into address
+func StringToAddress(value string) (vaa.Address, error) {
```

Comment:
These two functions seem to be getting replicated in a lot of different watchers. Should they be moved in to the common package?